core/payments/ondemand/on_demand_ledger.go

```diff
@@ -109,33 +147,60 @@ func (odl *OnDemandLedger) Debit(
 
 	blobCost := odl.computeCost(symbolCount)
 
-	newCumulativePayment, err := odl.cumulativePaymentStore.AddCumulativePayment(ctx, blobCost, odl.totalDeposits)
-	if err != nil {
-		return nil, fmt.Errorf("add cumulative payment: %w", err)
+	odl.lock.Lock()
+	defer odl.lock.Unlock()
+
+	newCumulativePayment := new(big.Int).Add(odl.cumulativePayment, blobCost)
+	if newCumulativePayment.Cmp(odl.totalDeposits) > 0 {
+		return nil, &InsufficientFundsError{
+			CurrentCumulativePayment: odl.cumulativePayment,
+			TotalDeposits:            odl.totalDeposits,
+			BlobCost:                 blobCost,
+		}
+	}
+
+	if odl.cumulativePaymentStore != nil {
```

Comment:
Idea, feel free to skip it if you don't like it.

If we add the following to the `cumulativePaymentStore` implementation's functions:

```
if s == nil {
   // sane no-op behavior
}
```

You could then remove an indention level from a bunch of code in this class, since calling functions on nil would not panic.

Reply:
77b258a4a6b7070fe00004b1d62ed4144c6383b6

Reply:
Note: I intentionally only added this to the `StoreCumulativePayment` method. `GetCumulativePayment` should will only be called in places where the pointer isn't nil